tree/treeplayer/test/treeprocs/treeprocessors.cxx

```diff
@@ -56,6 +59,42 @@ void DeleteFiles(const std::vector<std::string> &filenames)
       gSystem->Unlink(f.c_str());
 }
 
+class FilesRAII {
+   std::vector<std::string> fFileNames;
+
+public:
+   FilesRAII(const std::vector<std::string> &treenames, const std::vector<std::string> &filenames, int nEvts = 10)
+      : fFileNames(filenames)
+   {
+      WriteFiles(treenames, filenames, nEvts);
+   }
+   ~FilesRAII() { DeleteFiles(fFileNames); }
+};
+
+// See issue #15425
+TEST(TreeProcessorMP, moreWorkersThanEvents)
+{
+   auto func = [](TTreeReader &r){
+      int n=0;
+      while(r.Next()) n++;
+      auto par = new TParameter<int>("n", n);
+      return par;
+   };
+
+   std::vector<std::string> files = {"f_moreWorkersThanEvents.root"};
+   std::vector<std::string> trees = {"t"};
+
+   FilesRAII fr(trees, files, 2);
+
+   ROOT::TestSupport::CheckDiagsRAII checkDiag;
+   checkDiag.requiredDiag(kError, "TTreeProcessorMP::HandlePoolCode", "[E][C] a worker encountered an error: [S2]: could not set TTreeReader to range 2 1", false);
+
+   ROOT::TTreeProcessorMP proc(3);
+   auto res = proc.Process(files, func);
+
+   EXPECT_EQ(2, res->GetVal()) << "The counter incremented in the worker processes has the wrong value.";
```

Comment:
```suggestion
   EXPECT_EQ(2, res->GetVal());
```

I believe the ctest report will already tell that `2 != res->GetVal()`.

Reply:
Yes, if you want that potential error output is even clearer.